qutip/tests/test_entropy.py

```diff
@@ -89,6 +89,56 @@ def test_component_selection(self):
         assert abs(qutip.entropy_mutual(dm, [0, 2], [1]) - expect) < 1e-13
 
 
+class TestRelativeEntropy:
+    def test_rho_or_sigma_not_oper(self):
+        rho = qutip.ket("00")
+        sigma = qutip.ket("01")
+        with pytest.raises(TypeError) as exc:
+            qutip.entropy_relative(qutip.ket2dm(rho), sigma)
+        assert str(exc.value) == "Inputs must be density matrices."
+        with pytest.raises(TypeError) as exc:
+            qutip.entropy_relative(rho, qutip.ket2dm(sigma))
+        assert str(exc.value) == "Inputs must be density matrices."
+
+    def test_base_not_2_or_e(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = qutip.ket2dm(qutip.ket("01"))
+        with pytest.raises(ValueError) as exc:
+            qutip.entropy_relative(rho, sigma, base=3)
+        assert str(exc.value) == "Base must be 2 or e."
+
+    def test_infinite_relative_entropy(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = qutip.ket2dm(qutip.ket("01"))
+        assert qutip.entropy_relative(rho, sigma) == np.inf
+
+    def test_base_2_or_e(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = rho + qutip.ket2dm(qutip.ket("01"))
+        sigma = sigma.unit()
+        assert (
+            qutip.entropy_relative(rho, sigma) == pytest.approx(0.69314718)
+        )
+        assert (
+            qutip.entropy_relative(rho, sigma, base=np.e)
+            == pytest.approx(0.69314718)
+        )
+        assert qutip.entropy_relative(rho, sigma, base=2) == pytest.approx(1)
```

Comment:
Are there any other known cases we can test as well?

I'm not sure on best practices here - is testing the default argument of `base` something that we want to enforce via tests like this?

Also, minor nit, but perhaps the base-e comparisons could be to `np.log(2)` for clarity.

Reply:
Possible test cases: 
for pure states, either rho = sigma and S = 0, or rho ~= sigma and S = inf.
comparing pure states against the maximally mixed state should yield the classical rel entropy, e.g. |00> against 0.25(|00> + |01> + |10> + |11>) should return S = 2.
(I checked a few combinations and apart from the tolerance issue above for |0> and |+> I did get the correct results)

Reply:
@Albantakis Thanks! I added a test for the pure vs maximally mixed density matrix.

Reply:
@jakelishman It's nice to test that the defaults work as expected, and its easy to update the test if it changes (and if it changes accidentally it's probably a bug).